telegram/ext/filters.py

```diff
@@ -192,12 +195,118 @@ def filter(self, message):
     class _Document(BaseFilter):
         name = 'Filters.document'
 
+        class Category(BaseFilter):
+            """This Filter filters documents by their category in the mime-type attribute
+
+            Note:
+                This Filter only filters by the mime_type of the document,
+                    it doesn't check the validity of the document.
+                The user can manipulate the mime-type of a message and
+                    send media with wrong types that don't fit to this handler.
+
+            Examples:
+                Filters.documents.Category("audio/") returnes `True` for all types
+                of audio sent as file, for example "audio/mpeg" or "audio/x-wav"
+            """
+            name = "Filters.document.Category()"
+
+            def __init__(self, category):
+                """Initialize the category you want to filter
+
+                Args:
+                    category (str, optional): Category of the media you want to filter"""
+                self.category = category
+
+            def filter(self, message):
+                if message.document:
+                    return bool(message.document.mime_type.startswith(self.category))
+
+        application = Category(constants.MediaCategory.application)
```

Comment:
Could we perhaps expose these as `filters.document.category.application` instead of just `filters.document.application`?

Reply:
@bomjacob 
is this nesting really necessary? I think that `filters.document.application` is much more easier to use.